test/e2e/node_exporter_test.go

```diff
@@ -158,3 +158,69 @@ nodeExporter:
 		})
 	}
 }
+
+// This test ensures neccessary collectors stay operational after changing generic options in Node Exporter.
+func TestNodeExporterGenericOptions(t *testing.T) {
+	t.Cleanup(func() {
+		f.MustDeleteConfigMap(t, &corev1.ConfigMap{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      clusterMonitorConfigMapName,
+				Namespace: f.Ns,
+			},
+		})
+	})
+
+	collectorsToCheck := []string{
+		"cpu",
+		"diskstats",
+		"filesystem",
+		"hwmon",
+		"loadavg",
+		"meminfo",
+		"netclass",
+		"netdev",
+		"netstat",
+		"os",
+		"stat",
+		"time",
+		"uname",
+		"vmstat",
+	}
+
+	tests := []struct {
+		name   string
+		config string
+	}{
+		{
+			name:   "default config",
+			config: "",
+		},
+		{
+			name: "maxprocs = 1",
+			config: `
+nodeExporter:
+  maxProcs: 1`,
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.name, func(st *testing.T) {
+			f.MustCreateOrUpdateConfigMap(t, configMapWithData(t, test.config))
+
+			for _, nameCollector := range collectorsToCheck {
+
+				f.PrometheusK8sClient.WaitForQueryReturn(
+					t, 5*time.Minute, fmt.Sprintf(`min(node_scrape_collector_success{collector="%s"})`, nameCollector),
+					func(v float64) error {
+						if v != 1 {
+							return fmt.Errorf(`expecting min(node_scrape_collector_success{collector="%s"}) = 1 but got %v.`, nameCollector, v)
+						}
+						return nil
+					},
+				)
```

Comment:
I'm not sure checking a metric is a good test here. This might return a metric from before the config was applied.
Its probably better to get the node-exporter daemonset and make sure the argument was added to the node-exporter container arguments.

Reply:
The unit test already checks for the CLI args. I'd rather keep the end-to-end test more high-level and actually verify that the collectors are functional.

> This might return a metric from before the config was applied.

It shouldn't as the older metrics should be stale and not showing anymore.

Reply:
Ah yes my bad. Misread the earlier comment. In that case this lgtm!

Reply:
oops, I missed the latest messages, I removed the argument check in the e2e test and it needs a new `lgtm` label now 😅

Reply:
Sorry about that, my mistake!